app/controllers/api/automate_controller.rb

```diff
@@ -1,15 +1,14 @@
 module Api
   class AutomateController < BaseController
     def show
-      ae_browser = MiqAeBrowser.new(@auth_user_obj)
-      begin
-        resources = ae_browser.search(@req.c_suffix, ae_search_options)
-      rescue => err
-        raise BadRequestError, err.to_s
+      if browser_requested?
+        resources = collection_search
+        attributes = params['attributes'] ? %w(fqname) | params['attributes'].to_s.split(',') : nil
+        resources = resources.collect { |resource| resource.slice(*attributes) } if attributes.present?
+        render_resource :automate, :name => "automate", :subcount => resources.count, :resources => resources
+      else
+        super
```

Comment:
why not use the ae_browser for both types, /api/automate/:domain_name or /api/automate/:id ? default behavior we have with ae_browser is to just return the domain resource, it will be a consistent response coming back with the fqname, domain_fqname and such